src/Mailer/Mailer.php

```diff
@@ -224,58 +290,351 @@ public function __call(string $method, array $args)
-                ]);
-            }
+        if ($action === null) {
+            return $this->doSend('', $this->autoRender);
+        }
 
-            $this->_email->getMessage()->setHeaders($headers);
-            if (!$this->_email->viewBuilder()->getTemplate()) {
-                $this->_email->viewBuilder()->setTemplate($action);
-            }
+        if (!method_exists($this, $action)) {
+            throw new MissingActionException([
+                'mailer' => static::class,
+                'action' => $action,
+            ]);
+        }
 
+        $this->clonedInstances['message'] = clone $this->message;
+        $this->clonedInstances['renderer'] = clone $this->getRenderer();
+        if ($this->transport !== null) {
+            $this->clonedInstances['transport'] = clone $this->transport;
+        }
+
+        $this->getMessage()->setHeaders($headers);
+        if (!$this->viewBuilder()->getTemplate()) {
+            $this->viewBuilder()->setTemplate($action);
+        }
+
+        try {
             $this->$action(...$args);
 
-            $result = $this->_email->send();
+            $result = $this->deliver('');
         } finally {
-            $this->reset();
+            $this->restore();
         }
 
         return $result;
     }
 
     /**
-     * Reset email instance.
+     * Render content and set message body.
      *
+     * @param string $content Content.
      * @return $this
      */
-    protected function reset()
+    public function render(string $content = '')
     {
-        $this->_email = clone $this->_clonedEmail;
+        $content = $this->getRenderer()->render(
+            $content,
+            $this->message->getBodyTypes()
+        );
+
+        $this->message->setBody($content);
 
         return $this;
     }
 
+    /**
+     * Render content and send email using configured transport.
+     *
+     * @param string $content Content.
+     * @return array{headers: string, message: string}
+     */
+    public function deliver(string $content)
+    {
+        return $this->doSend($content);
+    }
+
+    /**
+     * Send email using configured transport and optionally log delivered message.
+     *
+     * @param string $content Content
+     * @param bool $render Flag to controller view rendering, default `true`.
+     * @return array{headers: string, message: string}
+     */
+    protected function doSend(string $content = '', $render = true)
+    {
+        if ($render) {
+            $this->render($content);
+        }
+
+        $result = $this->getTransport()->send($this->message);
+        $this->logDelivery($result);
+
+        return $result;
+    }
+
+    /**
+     * Set message body.
+     *
+     * Setting body content explicity will disable auto rendering.
+     *
+     * @param array $content Content array with keys "text" and/or "html" with
+     *   content string of respective type.
+     * @return $this
+     */
+    public function setBody(array $content)
```

Comment:
How do you see these new methods being used? Are they a longer term replacement for Email features? To me the separation between Mailer and Email was that Mailer applied more conventions and had more batteries-included for tasks like rendering templates. Having lower-level setters makes the purpose of each class a bit more murky.

Reply:
You are right, having this lower level method on `Mailer` does muddle things.

I added the method for setting message body mainly to control the auto rendering as I wanted a user to be able to directly set text/HTML content for the message and skip using `Renderer`. But I guess that's not necessary. A user can directly use `Message` and `Transport` in that case.

I have removed these methods.